spectrum-offchain-lm/src/bundle/rocksdb.rs

```diff
@@ -66,13 +66,14 @@ impl BundleRepo for BundleRepoRocksDB {
         let db = self.db.clone();
         spawn_blocking(move || {
             let prefix = epoch_index_prefix(pool_id, epoch_ix);
+            let mut readopts = ReadOptions::default();
+            readopts.set_iterate_range(rocksdb::PrefixRange(prefix.clone()));
             let mut acc = Vec::new();
-            // Note that restricting the following iterator to only `prefix` leads to dropped
-            // bundles.
-            let mut iter = db.iterator(IteratorMode::From(&*prefix, Direction::Forward));
+            let mut iter = db.iterator_opt(IteratorMode::From(&prefix, Direction::Forward), readopts);
             while let Some(Ok((key_bytes, _))) = iter.next() {
-                if let Some((pid, bundle_id, init_epoch_ix)) = destructure_epoch_index_key(&*key_bytes) {
-                    if pid == pool_id && init_epoch_ix <= epoch_ix && !acc.contains(&bundle_id) {
+                if let Some((pid, bundle_id, init_epoch_ix)) = destructure_epoch_index_key(&key_bytes) {
+                    assert_eq!(init_epoch_ix, epoch_ix);
+                    if pid == pool_id && !acc.contains(&bundle_id) {
```

Comment:
Here I just removed unnecessary condition `init_epoch_ix <= epoch_ix`.

Reply:
Are you sure it's unnecessary? I added it as a fix to some bug where init_epoch_ix was > than epoch_ix. Not sure if it's possible now.

Reply:
Yes, I'm sure. Because with this change the iterator is rigidly fixed on `epoch_ix` only. Note that I added an assertion on line 75. It's never failed throughout live testing.